q2_types/multiplexed_sequences/_transformer.py

```diff
@@ -0,0 +1,32 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2022, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import skbio.io
+
+from . import (MultiplexedFastaQualDirFmt,
+               MultiplexedSingleEndBarcodeInSequenceDirFmt)
+
+from ..plugin_setup import plugin
+
+
+@plugin.register_transformer
+def _1(df: MultiplexedFastaQualDirFmt) -> \
+     MultiplexedSingleEndBarcodeInSequenceDirFmt:
+    seqs = open(df.sequences.path_maker())
+    qual = open(df.quality.path_maker())
+
+    result = MultiplexedSingleEndBarcodeInSequenceDirFmt()
+
+    # TODO: can I access the file name (forward.fastq.gz) from result
```

Comment:
I'm not sure if that's possible, but this doesn't seem terribly unreasonable to me to leave here. @ebolyen what do you think?

Reply:
There's really not a good way to do this at the moment. In any case, the specific filenames should be considered stable/part of the interface of the format itself, so I don't think the duplication is a problem.

(It's not any different from knowing that some method is named `object.some_specific_interface()`, and we wouldn't really worry about duplicating knowledge of that.)